### PR TITLE
Move more LA steps under conditional

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1 [hunk 1]
```diff
@@ -479,8 +479,8 @@ az role assignment create --assignee $systemIdentityPrincipal --role "Storage Bl
 # Uploading root CA certificate to IoT hub and verifying
 ##################################################################################################################################
 
-$certExits = az iot hub certificate list -g $ResourceGroup --hub-name $iotHubName --query "value[?name=='$hubUploadCertificateName']" --output tsv
-if ($certExits)
+$certExists = az iot hub certificate list -g $ResourceGroup --hub-name $iotHubName --query "value[?name=='$hubUploadCertificateName']" --output tsv
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:482]: Fix misspelling

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1 [hunk 2]
```diff
@@ -653,24 +653,24 @@ az iot dps enrollment create `
 
 if ($EnableIotHubSecuritySolution)
 {
-    Write-Host "`nCreating a self-signed certificate and placing it in $ResourceGroup."
+    Write-Host "`nCreating a self-signed certificate lor LA and placing it in $ResourceGroup."
     az ad app credential reset --id $logAnalyticsAppId --create-cert --keyvault $keyVaultName --cert $ResourceGroup --output none
     Write-Host "`nSuccessfully created a self signed certificate for your application $logAnalyticsAppRegnName in $keyVaultName key vault with cert name $ResourceGroup."
-}
 
-Write-Host "`nFetching the certificate binary."
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:661]: This stuff should have been inside the new conditional on line 654. The console writes don't make it evident, and using a secret name of the rg for the LA app name is an odd choice. I didn't realize it was all part of the same logic, but now I see it is.

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1 [hunk 3]
```diff
@@ -741,6 +739,8 @@ $keyvaultKvps = @{
 if ($EnableIotHubSecuritySolution)
 {
     $keyvaultKvps.Add("LA-WORKSPACE-ID", $workspaceId)
+    $keyvaultKvps.Add("LA-AAD-APP-CERT-BASE64", $fileContentB64String)
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:742]: These are also optional.

**Review by abhipsaMisra — APPROVED** [review]

**Review by andyk-ms — APPROVED** [review]

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1 [hunk 4]
```diff
@@ -653,24 +653,24 @@ az iot dps enrollment create `
 
 if ($EnableIotHubSecuritySolution)
 {
-    Write-Host "`nCreating a self-signed certificate and placing it in $ResourceGroup."
+    Write-Host "`nCreating a self-signed certificate lor LA and placing it in $ResourceGroup."
```
Comment (azabbasi) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:656]:
```suggestion
    Write-Host "`nCreating a self-signed certificate for LA and placing it in $ResourceGroup."
```

**Review by azabbasi — APPROVED** [review]

**Pull request merged**